source/exe/main_common.cc

```diff
@@ -46,15 +46,18 @@ MainCommonBase::MainCommonBase(const Server::Options& options, Event::TimeSystem
                                ListenerHooks& listener_hooks,
                                Server::ComponentFactory& component_factory,
                                std::unique_ptr<Random::RandomGenerator>&& random_generator,
-                               Thread::ThreadFactory& thread_factory,
-                               Filesystem::Instance& file_system,
                                std::unique_ptr<ProcessContext> process_context)
-    : options_(options), component_factory_(component_factory), thread_factory_(thread_factory),
-      file_system_(file_system), stats_allocator_(symbol_table_) {
+    : options_(options), component_factory_(component_factory), stats_allocator_(symbol_table_) {
   // Process the option to disable extensions as early as possible,
   // before we do any configuration loading.
   OptionsImpl::disableExtensions(options.disabledExtensions());
 
+  // Enable core dumps as early as possible.
+  if (options_.coreDumpEnabled()) {
+    auto ret = platform_impl_.enableCoreDump();
```

Comment:
If it wasn't actually enabled can we fail and say it's not supported?

Reply:
What would failing mean here? Throwing an exception and exiting? Hmm I am somehow think it should be best effort...

Reply:
Yeah fair enough. A soft failure is fine. Can we at least split the logic and warn if it can't be enabled vs. info for enabled?

/wait